src/Workspaces/Core/Portable/Workspace/TextExtensions.cs

```diff
@@ -73,6 +73,50 @@ public static ImmutableArray<Document> GetRelatedDocumentsWithChanges(this Sourc
             return null;
         }
 
+        /// <summary>
+        /// Gets the <see cref="AdditionalDocument"/> from the corresponding workspace's current solution that is associated with the source text's container 
+        /// in its current project context, updated to contain the same text as the source if necessary.
+        /// </summary>
+        public static TextDocument? GetOpenAdditionalDocumentInCurrentContextWithChanges(this SourceText text)
+        {
+            if (Workspace.TryGetWorkspace(text.Container, out var workspace))
+            {
+                var solution = workspace.CurrentSolution;
+                var id = workspace.GetDocumentIdInCurrentContext(text.Container);
+                if (id == null || !solution.ContainsAdditionalDocument(id))
+                {
+                    return null;
+                }
+
+                return solution.WithAdditionalDocumentText(id, text, PreservationMode.PreserveIdentity)
```

Comment:
I also thought so, but unlike the `WithDocumentText` API set, which has an overload for multiple document IDs, the `WithAdditionalDocumentText` and `WithAnalyzerConfigDocumentText` does not have an overload for multiple document IDs. https://github.com/dotnet/roslyn/blob/c44c7e89a8cab8c9862f710b62a745daf0e26c9a/src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs#L1738-L1741
https://github.com/dotnet/roslyn/blob/main/src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs#L1391-L1395
https://github.com/dotnet/roslyn/blob/c44c7e89a8cab8c9862f710b62a745daf0e26c9a/src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs#L1418-L1422

Reply:
Filed https://github.com/dotnet/roslyn/issues/64701 to do a more thorough audit of places where we need to work on all linked documents instead of a single document.